packages/cli/src/ui/contexts/UIStateContext.tsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { createContext, useContext } from 'react';
+import {
+  HistoryItem,
+  StreamingState,
+  ThoughtSummary,
+  ConsoleMessageItem,
+  ShellConfirmationRequest,
+  ConfirmationRequest,
+  HistoryItemWithoutId,
+} from '../types.js';
+import { type CommandContext, type SlashCommand } from '../commands/types.js';
+import { TextBuffer } from '../components/shared/text-buffer.js';
+import { IdeContext } from '@google/gemini-cli-core';
+
```

Comment:
UIStateContext worries me a bit. This is a ton of info to be promoting to a context. Is this really needed? Would be better to explicitly pass these around where needed with prop drilling unless there are reasons we really need to do this to achieve this refactor.

Reply:
sounds good.